src/lucky/routable.cr

```diff
@@ -282,29 +282,37 @@ module Lucky::Routable
     {% end %}
     anchor : String? = nil
     ) : Lucky::RouteHelper
-      path = path_from_parts(
-        {% for param in path_params %}
-          {{ param.gsub(/:/, "").id }},
-        {% end %}
-        {% for param in optional_path_params %}
-          {{ param.gsub(/^\?:/, "").id }},
+      path = String.build do |io|
+        path_from_parts(
+          io,
+          {% for param in path_params %}
+            {{ param.gsub(/:/, "").id }},
+          {% end %}
+          {% for param in optional_path_params %}
+            {{ param.gsub(/^\?:/, "").id }},
+          {% end %}
+        )
+        query_params = {} of String => String
+        {% for param in PARAM_DECLARATIONS %}
+          # add query param if given and not nil
+          query_params["{{ param.var }}"] = {{ param.var }}.to_s unless {{ param.var }}.nil?
         {% end %}
-      )
-      query_params = {} of String => String
-      {% for param in PARAM_DECLARATIONS %}
-        # add query param if given and not nil
-        query_params["{{ param.var }}"] = {{ param.var }}.to_s unless {{ param.var }}.nil?
-      {% end %}
-      unless query_params.empty?
-        path += "?#{HTTP::Params.encode(query_params)}"
-      end
+        unless query_params.empty?
+          io << '?'
+          {% if compare_versions(Crystal::VERSION, "1.10.0") < 0 %}
```

Comment:
Should this emit a deprecated message somehow? Code like this probably shouldn't live forever.

Reply:
True. The deprecation is an annotation though.. Oh, I could just move the call in to a separate method with the deprecation. Ok, I'll look at doing that.

Reply:
As of `1.9.0` there is `ASTNode#warning` you can use to handle it manually. `@[Deprecated]` basically does the same thing.

Reply:
oh! I didn't know. That's great. Thanks for the tip @Blacksmoke16 :raised_hands: